File: ShipManagement/ShipApp/src/app/Models/trip-particulars.ts
export class TripParticulars {
    tripParticularId: number | undefined;
    tripId :number|undefined;
    tripDate: Date | undefined;
    fishId: number | undefined;
    ratePerKg :number |undefined;
    totalWeight:number|undefined;
    amount :number |undefined;
    createdDate: Date | undefined;
    updatedDate: Date | undefined;
    boatId: number | undefined;
  

    /**
     *
     */
    constructor(tpid?: number, 
                tname?:string,
                tid?: number, 
                tdate?: Date, 
                fid?: number, 
                rate?: number, 
                wt?: number,
                amt?: number,
                cdate?: Date, 
                udate?: Date, 
                bid?: number) 
    {
        this.tripParticularId = tpid ?? 0;
        //this.tripName =tname ?? undefined;
        this.tripId = tid ?? 0;
        this.tripDate = tdate ?? undefined
        this.fishId = fid ?? 0;
        this.ratePerKg = rate ?? 0;
        this.totalWeight = wt ?? 0;
        this.amount = amt ?? 0;
        this.createdDate = cdate ?? undefined;
        this.updatedDate = udate ?? undefined;
        this.boatId = bid ?? 0;
    }
}

File: ShipmanagementAPI/DataLayer/Entities/TripParticulars.cs
using System.ComponentModel.DataAnnotations.Schema;

namespace DataLayer.Entities
{
    public class GoodsDetails
    {
        public string FishName { get; set; }
        public decimal? RatePerKg { get; set; } = 0;
        public decimal? TotalWeight { get; set; } = 0;
        public decimal Amount { get; set; }

        public GoodsDetails(string fishname, decimal rateperkg, decimal totalweight, decimal amount)
        {
            this.FishName = fishname;
            this.RatePerKg = rateperkg;
            this.TotalWeight = totalweight;
            if (this.RatePerKg != 0 && totalweight != 0)
            {
                this.Amount = (decimal)(this.RatePerKg * this.TotalWeight);
            }
            else
            {
                this.Amount = amount;
            }
        }
    }
    public class TripParticular
    {
        public required int TripParticularId { get; set; }
        public required DateTime TripDate { get; set; }
        //public List<GoodsDetails>? TripParticulars { set; get; }

        // Properties previously in GoodsDetails
        public int FishId { get; set; }       // Name of the fish
        public decimal RatePerKg { get; set; } = 0;  // Rate per kilogram
        public decimal TotalWeight { get; set; } = 0;  // Total weight of goods
        public decimal Amount { get; set; }        // Amount calculated based on weight and rate
        public required DateTime CreatedDate { get; set; }
        public required DateTime UpdatedDate { get; set; }
        /*
         * TripInfo to TripParticular (One-to-Many):
         *      Each TripInfo can have multiple TripParticular records.
         *      Each TripParticular is associated with one TripInfo.
         *      This is also a one-to-many relationship.
         */
        // Foreign key for TripInfo
        [ForeignKey("TripId")]
        public int TripId { get; set; }
        public TripInformation TripInfo { get; set; }

    }
}
